src/styles/text/text.js

```diff
@@ -0,0 +1,371 @@
+            let style_key = this.constructStyleKey(style);
+
+            if (!this.texts[tile][style_key]) {
+                this.texts[tile][style_key] = {};
+            }
+
+            if (!this.geometries[tile][style_key]) {
+                this.geometries[tile][style_key] = feature.geometry;
+            }
+
+            this.texts[tile][style_key][text] = {
+                text_style: style
+            };
+
+            if (rule.labels) {
+                this.label_style = rule.labels;
+            }
+
+            this.geometries[tile][style_key][text] = feature.geometry;
+        }
+
+        tile_data.queue.push([feature, rule, context, tile_data]);
+    },
+
+    constructFontStyle (rule) {
+        let style = this.font_style;
+
+        if (rule.font) {
+            style = {
+                typeface: rule.font.typeface || this.font_style.typeface,
+                size: rule.font.size || this.font_size.font_size,
+                fill: rule.font.fill === undefined ? this.font_style.fill : Utils.toCanvasColor(rule.font.fill),
+                stroke: rule.font.stroke === undefined ? this.font_style.stroke : Utils.toCanvasColor(rule.font.stroke)
+            };
+        }
+
+        return style;
+    },
+
+    constructStyleKey ({ typeface, size, fill, stroke }) {
+        return `${typeface}/${size}/${fill}/${stroke}`;
+    },
+
+    build (style, vertex_data) {
+        var vertex_template = this.makeVertexTemplate(style);
+
+        Builders.buildSpriteQuadsForPoints(
+            [ style.label.position ],
+            Utils.scaleInt16(style.label.size[0], 128), Utils.scaleInt16(style.label.size[1], 128),
+            Utils.scaleInt16(Utils.radToDeg(style.label.angle), 360),
+            Utils.scaleInt16(style.scale, 256),
+            vertex_data,
+            vertex_template,
+            this.vertex_layout.index.a_shape,
+            { texcoord_index: this.vertex_layout.index.a_texcoord, texcoord_scale: this.texcoord_scale }
+        );
+    },
+
+    buildLines (lines, style, vertex_data) {
+        if (style.discarded) {
+            return;
+        }
+
+        this.build(style, vertex_data);
+    },
+
+    buildPoints (points, style, vertex_data) {
+        if (style.discarded) {
+            return;
+        }
+
+        this.build(style, vertex_data);
+    },
+
+    _parseFeature (feature, rule_style, context) {
+        let style = this.feature_style;
+        let tile = context.tile.key;
+        let text = feature.properties.name;
+
+        let font_style = this.constructFontStyle(rule_style);
+        let style_key = this.constructStyleKey(font_style);
+
+        let discarded = this.texts[tile][style_key][text] === undefined;
+
+        style.discarded = discarded;
+
+        if (!discarded) {
+            let text_info = this.texts[tile][style_key][text];
+            this.texcoord_scale = text_info.texcoords;
+
+            style.text = text;
+
+            style.angle = rule_style.angle || 0;
```

Comment:
I think this is leftover from the old sprites implementation, now that the `Label` class is setting the angle, right?